### DISCUSSION TITLE
[RTK Query] Ignoring serialization for specific query endpoints

### DISCUSSION THREAD

**Discussion by chwallen** [reduxjs/redux-toolkit#3003] [Q&A] [answered]
We have several endpoints with a variable argument where each can provide up to several MB of data. With the serialization middleware enabled, the dev environment is very slow.

How can we selectively ignore serialization for specific endpoints in our API? We can do a very crude ignore with `ignoredPaths: [api.reducerPath]`. Trying to do something more specific does not work as the paths at `api.reducerPath.queries` requires that we know its arguments ahead of time (which we don't).

**Comment by markerikson** [comment, marked as answer]
Hmm. At the moment I'm not sure there's a better approach.  There _is_ an open PR to add regex support, and maybe we could add a callback form or something.

**Reply by chwallen** [reply]
As I suspected, thanks! It would be a "nice to have" if we could keep it enabled for other endpoints, but not a deal breaking by any means. :)

The docs could be updated to state that it is not possible to do with Query (yet?).

**Reply by phryneas** [reply]
It is probably not a necessity if you use RTK Query normally with a `baseQuery`. Either all your endpoints will contain non-serializable data, or none will - if you use the provided `fetchBaseQuery` and don't do any weird things in `transformResult`, you are pretty much guaranteed that the data is okay. So you can just turn off that middleware for the whole api.

**Reply by chwallen** [reply]
On a few occations, we have accidentally introduced non-serializable objects (mainly dates from our date library) via `transformResult`/`queryFn`. In those cases, the middleware has helped us to immediately spot it. In general though, you are absolutely correct that we could just disable the middleware for the API, which is what we have opted to do.
